From f8d9f26dd3de6e954987d6830f23def17938e21e Mon Sep 17 00:00:00 2001
From: Xu Bing <bing.xu@intel.com>
Date: Thu, 19 Jun 2025 15:06:51 +0800
Subject: [PATCH] Fix connection errors when hot pluging USB camera

system will throw CAMERA_ERROR_UNKNOWN error when hot pluging USB
camera, It's a fatal error but Camera2 app think it's a normal
error, correct it.

Tracked-On: OAM-132658
Signed-off-by: Xu Bing <bing.xu@intel.com>
---
 src/com/android/camera/CameraActivity.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/camera/CameraActivity.java b/src/com/android/camera/CameraActivity.java
index 3133f57eb..759f6664b 100644
--- a/src/com/android/camera/CameraActivity.java
+++ b/src/com/android/camera/CameraActivity.java
@@ -1366,6 +1366,10 @@ public class CameraActivity extends QuickActivity
                 public void onCameraError(int errorCode) {
                     // Not a fatal error. only do Log.e().
                     Log.e(TAG, "Camera error callback. error=" + errorCode);
+                    if (errorCode == 1) {
+                        Log.e(TAG, "Camera error unknown, throw fatal error to reset!");
+                        onFatalError();
+                    }
                 }
                 @Override
                 public void onCameraException(
-- 
2.34.1

